PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/test/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/DisableAutoCompleteCheckTest.java

```diff
@@ -0,0 +1,203 @@
+
+import com.intellij.codeInspection.ProblemsHolder;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiDeclarationStatement;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiReferenceExpression;
+import com.intellij.psi.PsiType;
+import com.intellij.psi.PsiVariable;
+import com.microsoft.azure.toolkit.intellij.azure.sdk.buildtool.DisableAutoCompleteCheck.DisableAutoCompleteVisitor;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+import static org.mockito.ArgumentMatchers.contains;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+/**
+ * This class is used to test the DisableAutoCompleteCheck class.
+ * It tests the visitDeclarationStatement method of the DisableAutoCompleteVisitor class.
+ */
+public class DisableAutoCompleteCheckTest {
+
+    // Create a mock ProblemsHolder to be used in the test
+    @Mock
+    private ProblemsHolder mockHolder;
+
+    // Create a mock JavaElementVisitor for visiting the PsiDeclarationStatement
+    @Mock
+    private JavaElementVisitor mockVisitor;
+
+    // Create a mock PsiDeclarationStatement.
+    @Mock
+    private PsiDeclarationStatement mockDeclarationStatement;
+
+
+    @BeforeEach
+    public void setUp() {
+        // Set up the test
+        mockHolder = mock(ProblemsHolder.class);
+        mockVisitor = createVisitor();
+        mockDeclarationStatement = mock(PsiDeclarationStatement.class);
+    }
+
+    /**
+     * Test the visitDeclarationStatement method of the DisableAutoCompleteVisitor class.
+     * This test checks if the auto-complete feature is disabled for the ServiceBusReceiverClient.
+     * If the auto-complete feature is not disabled, a problem is registered with the ProblemsHolder.
+     */
+    @Test
+    public void testProblemRegisteredWithoutDisableAutoComplete() {
+
+        String packageName = "com.azure";
+        String clientName = "ServiceBusReceiverClient";
+        int numOfInvocations = 1;
+        String methodFound = "notDisableAutoComplete";
+
+
+        // Assert
+        verifyProblemRegistered(packageName, clientName,numOfInvocations, methodFound);
+    }
+
+    /**
+     * Test the visitDeclarationStatement method of the DisableAutoCompleteVisitor class.
+     * This test checks if the auto-complete feature is disabled for the ServiceBusProcessorClient.
+     * If the auto-complete feature is not disabled, a problem is registered with the ProblemsHolder.
+     */
+    @Test
+    public void testProblemRegisteredWithoutDisableAutoCompleteWithAnotherServiceBusClientToCheck() {
+
+        String packageName = "com.azure";
+        String clientName = "ServiceBusProcessorClient";
+        int numOfInvocations = 1;
```

Comment:
Should this be 0?
because the method is not equal to `disableAutoComplete`?

Reply:
if its NOTdisableAutoComplete then a problem should be registered -- we're looking for the asbence of the client